command/apply.go

```diff
@@ -272,6 +274,8 @@ Options:
                          modifying. Defaults to the "-state-out" path with
                          ".backup" extension. Set to "-" to disable backup.
 
+  -lock-state=true       Lock the state file when locking is supported.
```

Comment:
Bike shed: Let's just use `-lock`. Maybe we'll lock more in the future maybe we won't but I think its clear regardless and I'd prefer the aesthetic of it.